redundant-bmc/src/manager.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <sdbusplus/async.hpp>
+#include <xyz/openbmc_project/State/BMC/Redundancy/server.hpp>
+
+namespace rbmc
+{
+
+using RedundancyIntf =
+    sdbusplus::xyz::openbmc_project::State::BMC::server::Redundancy;
+using RedundancyInterface = sdbusplus::server::object_t<RedundancyIntf>;
+
+/**
+ * @class Manager
+ *
+ * Manages the high level operations of the redundant
+ * BMC functionality.
+ */
+class Manager
+{
+  public:
+    Manager() = delete;
+    ~Manager() = default;
+    Manager(const Manager&) = delete;
+    Manager& operator=(const Manager&) = delete;
+    Manager(Manager&&) = delete;
+    Manager& operator=(Manager&&) = delete;
+
+    /**
+     * @brief Constructor
+     *
+     * @param ctx - The async context object
+     */
+    explicit Manager(sdbusplus::async::context& ctx);
+
+  private:
+    /**
+     * @brief Kicks off the Manager startup
+     */
+    sdbusplus::async::task<> startup();
+
+    /**
+     * @brief Emits a heartbeat signal every second
+     *
+     * The sibling gets this via the sibling app to
+     * know all's well.
+     */
+    sdbusplus::async::task<> doHeartBeat();
+
+    /**
+     * @brief The async context object
+     */
+    sdbusplus::async::context& ctx;
+
+    /**
+     * @brief The Redundancy D-Bus interface
+     */
+    RedundancyInterface redundancyInterface;
```

Comment:
Just curious, why did you choose composition for the DBus server class?

Reply:
Inheritance is only required when you have a method you need to implement or you need to override the behavior of a property get/set, which isn't the case here.

Reply:
Right, so we’ll need to override the Failover method, correct? Not now, but when we plan to introduce this method as outlined in the [design doc](https://gerrit.openbmc.org/c/openbmc/docs/+/70233/9/designs/redundant-bmc.md#363), right?

Reply:
yea, there will have to be some derived class to implement the Failover method.  It can't be Manager though if I don't want it on the active BMC.  Leaning towards putting it on its own interface.

Reply:
Oh, I see. You're planning to create a separate class.